test/org/sagebionetworks/bridge/validators/SurveyValidatorTest.java

```diff
@@ -380,6 +383,84 @@ public void willValidateRuleReferencesToNonQuestions() {
+        String json = TestUtils.createJson("{'operator':'eq','value':'No',"+
+                "'skipTo':'theend','endSurvey':true}");
+        SurveyRule rule = BridgeObjectMapper.get().readValue(json, SurveyRule.class);
+        constraints.getRules().add(rule);
+        
+        SurveyQuestion question = new DynamoSurveyQuestion();
+        question.setIdentifier("start");
+        question.setUiHint(UIHint.TEXTFIELD);
+        question.setPrompt("Prompt");
+        question.setConstraints(constraints);
+        survey.getElements().add(question);
+        
+        SurveyInfoScreen info = new DynamoSurveyInfoScreen();
+        info.setTitle("Title");
+        info.setPrompt("Prompt");
+        info.setIdentifier("theend");
+        survey.getElements().add(info);
+
+        try {
+            Validate.entityThrowingException(validator, survey);
+            fail("Should have thrown exception");
+        } catch (InvalidEntityException e) {
+            assertEquals("rule cannot have a skipTo target and an endSurvey property", errorFor(e, "elements[0].rule"));
+        }
+    }
+    
+    @Test
+    public void willVerifyRuleHasEitherSkipToTargetOrEndSurvey() throws Exception {
+        Survey survey = new DynamoSurvey();
+        survey.setName("Name");
+        survey.setIdentifier("Identifier");
+        survey.setStudyIdentifier("study-key");
+        survey.setGuid("guid");
+        
+        StringConstraints constraints = new StringConstraints();
+        
+        // This is actually the only way to create an invalid rule (deserializing JSON).
+        String json = TestUtils.createJson("{'operator':'eq','value':'No'}");
+        SurveyRule rule = BridgeObjectMapper.get().readValue(json, SurveyRule.class);
+        constraints.getRules().add(rule);
+        
+        SurveyQuestion question = new DynamoSurveyQuestion();
+        question.setIdentifier("start");
+        question.setUiHint(UIHint.TEXTFIELD);
+        question.setPrompt("Prompt");
+        question.setConstraints(constraints);
+        survey.getElements().add(question);
+
+        try {
+            Validate.entityThrowingException(validator, survey);
+            fail("Should have thrown exception");
+        } catch (InvalidEntityException e) {
+            assertEquals("rule must have a skipTo target or an endSurvey property", errorFor(e, "elements[0].rule"));
+        }
+    }
+    
+    @Test
+    public void cannotSetEndSurveyToFalse() throws Exception {
+        String json = TestUtils.createJson("{'operator':'eq','value':'No',"+
+                "'skipTo':'theend','endSurvey':false}");
+        SurveyRule rule = BridgeObjectMapper.get().readValue(json, SurveyRule.class);
+        assertNull(rule.getEndSurvey());
+        assertEquals(Operator.EQ, rule.getOperator());
+        assertEquals("theend", rule.getSkipToTarget());
+        assertEquals("No", rule.getValue());
+    }
```

Comment:
Should this test be in SurveyRuleTest instead? The logic in question being tested is in SurveyRule, not SurveyValidator?